torch/nn/utils/rnn.py

```diff
@@ -26,7 +26,8 @@ class PackedSequence(PackedSequence_):
         batch_sizes (list[int]): list of integers holding information about
             the batch size at each sequence step
     """
-    pass
+    def cuda(self):
+        return type(self)(self.data.cuda(), self.batch_sizes)
```

Comment:
Can you add `cpu`, `float`, `double` and `half` too? Might be a good idea to implement them like this:

```python
def _make_cast(name):
  def fn(self, *args, **kwargs):
    return type(self)(getattr(self.data, name)(*args, **kwargs), self.batch_sizes)
  return fn

cuda = _make_cast('cuda')
cpu = _make_cast('cpu')
```

Reply:
Good one, I didn't think of the cuda args.  I really like the syntax, however:
```
import time
num = 10**7

class foo:
    bar = 5

f = foo()

t0 = time.time()
[f.bar  for _ in range(num)]
print('f.bar :', (time.time()-t0)/num)

t0 = time.time()
[getattr(f, 'bar') for _ in range(num)]
print('getattr(f,"bar") :', (time.time()-t0)/num)

> f.bar : 5.444774627685547e-08
> getattr(f,"bar") : 1.235971450805664e-07
```
Questions:
(1) Do you think someone will want to call this more than once per batch?
(2) Also what do you think of a generalization `_gen_cast_of(attr, name)`?

Reply:
I don't think the difference matters this much in here, but I'm fine with repeating all the cases manually too. Can you explain what do you mean in (2)?